File: Python/606. Construct String from Binary Tree.py
class Solution:
    def tree2str(self, root: Optional[TreeNode]) -> str:
        
        self.ans =''
        self.dfs(root)
        return self.ans[1:-1]
        
        
        
    def dfs(self,root):
        if root == None:
            return 
        
        self.ans += '(' + str(root.val)
        if root.left == None and root.right != None:
            self.ans += '()'
            
        self.dfs(root.left)
        self.dfs(root.right)
        self.ans += ')'
        return

File: Java/606. Construct String from Binary Tree.java
class Solution {
    String ans = "";
    public String tree2str(TreeNode root) {
        dfs(root);
        return ans.substring(1,ans.length()-1);
    }
    void dfs(TreeNode root){
        if(root==null)
            return;
        ans+="(" + Integer.toString(root.val);
        if(root.left==null && root.right!=null)
            ans += "()";
        dfs(root.left);
        dfs(root.right);
        ans += ")";
        return;
    }
}